### PR TITLE
Connectivity Tests: Mitigate hanging support bundle

### PR DESCRIPTION
**Pull request by and-rewsmith**
Retries for support bundle to unblock connectivity tests in dashboard. We need separate investigation to mitigate in product.

Changes:
1. Disable log upload
2. Timeout and retry support bundle

## Azure IoT Edge PR checklist:

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines and Best Practices
- [x] I have read the [contribution guidelines](https://github.com/azure/iotedge#contributing).
- [x] Title of the pull request is clear and informative.
- [x] Description of the pull request includes a concise summary of the enhancement or bug fix.

### Testing Guidelines
- [x] Pull request includes test coverage for the included changes.
- Description of the pull request includes 
	- [x] concise summary of tests added/modified
	- [ ] local testing done.  

### Draft PRs
- Open the PR in `Draft` mode if it is:
	- Work in progress or not intended to be merged.
	- Encountering multiple pipeline failures and working on fixes.

_Note: We use the kodiakhq bot to merge PRs once the necessary checks and approvals are in place. When it merges a PR, kodiakhq converts the PR title to the commit title, PR description to the commit description, and squashes all the commits in the PR to a single commit. The net effect is that entire PR becomes a single commit. Please follow the best practices mentioned [here](https://chris.beams.io/posts/git-commit/#:~:text=The%20seven%20rules%20of%20a%20great%20Git%20commit,what%20and%20why%20vs.%20how%20For%20example%3A%20) for the PR title and description_

### CODE REVIEW

File: scripts/linux/trcE2ETest.sh [hunk 1]
```diff
@@ -296,18 +296,38 @@ function print_deployment_logs() {
...
     mkdir -p $working_folder/support
     time=$(echo $test_start_time | sed 's/ /T/' | sed 's/$/Z/')
-    iotedge support-bundle -o $working_folder/support/iotedge_support_bundle.zip --since "$time"
-    print_highlighted_message "Finished getting support Bundle Logs"
+
+    MAX_RETRIES=4
+    RETRY_COUNT=0
+    DID_TIMEOUT=false
+
+    while [ "$RETRY_COUNT" -lt "$MAX_RETRIES" ] && [ "$DID_TIMEOUT" = false ]; do
```
Comment (damonbarry) [inline scripts/linux/trcE2ETest.sh:307]: If I'm reading this logic correctly, we'll only retry once. If there's a timeout, then `DID_TIMEOUT` will be set to true and the while condition above will fail.

Comment (and-rewsmith) [reply scripts/linux/trcE2ETest.sh:307]: Will fix rushed through this part.

**Comment by and-rewsmith** [comment]
I previously had a very simple singular retry but am now doing multiple retries with max retry count. Testing these changes:
https://msazure.visualstudio.com/One/_build/results?buildId=71290637&view=results

File: e2e_deployment_files/connectivity_deployment.template.json [hunk 2]
```diff
@@ -347,7 +347,7 @@
             "version": "1.0",
             "type": "docker",
             "status": "running",
-            "restartPolicy": "always",
+            "restartPolicy": "never",
```
Comment (damonbarry) [inline e2e_deployment_files/connectivity_deployment.template.json:350]: Was this change intentional?

**Review by damonbarry — APPROVED** [review]

File: e2e_deployment_files/connectivity_deployment.template.json (hunk 2 above)
Comment (and-rewsmith) [reply e2e_deployment_files/connectivity_deployment.template.json:350]: Yes it isn't designed to restart and still function. I encountered some confusing behavior when debugging this so was trying to clean this up.
